Sources/FeaturevisorSDK/Instance.swift

```diff
@@ -81,6 +99,46 @@ public struct Evaluation {
         self.variableValue = variableValue
         self.variableSchema = variableSchema
     }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(featureKey, forKey: .featureKey)
+        try container.encode(reason.rawValue, forKey: .reason)
+        try container.encode(bucketValue, forKey: .bucketValue)
+        try container.encode(ruleKey, forKey: .ruleKey)
+        if let error = error {
+            try container.encode(error.localizedDescription, forKey: .error)
+        }
+        try container.encode(enabled, forKey: .enabled)
+        try container.encode(traffic, forKey: .traffic)
+        try container.encode(sticky, forKey: .sticky)
+        try container.encode(initial, forKey: .initial)
+        try container.encode(variation, forKey: .variation)
+        try container.encode(variationValue, forKey: .variationValue)
+        try container.encode(variableKey, forKey: .variableKey)
+        try container.encode(variableValue, forKey: .variableValue)
+        try container.encode(variableSchema, forKey: .variableSchema)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        featureKey = try container.decode(FeatureKey.self, forKey: .featureKey)
+        reason = try EvaluationReason(rawValue: container.decode(String.self, forKey: .reason)) ?? .error
+        bucketValue = try container.decodeIfPresent(BucketValue.self, forKey: .bucketValue)
+        ruleKey = try container.decodeIfPresent(RuleKey.self, forKey: .ruleKey)
+        error = try container.decodeIfPresent(String.self, forKey: .error).flatMap { NSError(domain: "", code: 0, userInfo: [NSLocalizedDescriptionKey: $0]) }
```

Comment:
Error is optional, so leave it as optional and don't create "dummy" error for it. It's confusing.